### DISCUSSION TITLE
Option to change the directory for API Routes

### DISCUSSION THREAD

**Discussion by siamekanto19** [vercel/next.js#37947] [Ideas]
### Describe the feature you'd like to request

Currently, to create API Routes, we have to put those files under the `pages` directory inside `api` folder. Having all the APIs nested inside the `pages` directory makes it a hassle to jump back and forth pages and APIs. Often times, I get lost in the nesting while looking for a specific file inside the `pages` folder. I'm sure a lot of other developers also feel the same problem. 

This is not the end of the world, but if there was any option to change where the `api` folder belongs, it could be better for most of the developers working with Nextjs. Probably we want to get the `api` folder out of the `pages` directory and put it directly in the project's root. Having that kind of flexibility would make the Developer Experience a lot better. Some other frameworks like Nuxtjs already does it with their REST API system.

### Describe the solution you'd like

An option to change where the `api` directory belongs. Probably some kind of property inside `next.config.js` that indicates where the folder should be. Developers can configure this as per their requirements.

Example: 

```
module.exports = {
   apiDirectory:  "pages/api"     // Can be any absolute path
}
```

### Describe alternatives you've considered

I haven't actually found any other alternatives for this issue. I have found some remapping configs for the API, but that is not what I am looking for exactly.

**Comment by balazsorban44** [comment]
Duplicate of #37011
